### ISSUE TITLE
Handle grouping variables with duplicated elements

### ISSUE THREAD

**Issue by wlandau** [ropensci/drake#690]
As a user, I might expect to get a plan with 4 targets from this:

``` r
library(drake)
drake_plan(
  trace = TRUE,
  a = target(x, transform = map(x = c(1, 1))),
  b = target(f(a), transform = map(a))
)
#> # A tibble: 2 x 4
#>   target command x     a    
#>   <chr>  <chr>   <chr> <chr>
#> 1 a_1    1       1     a_1  
#> 2 b      f(a)    <NA>  <NA>
```

<sup>Created on 2019-01-22 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

We should either throw an error or add a new grouping variable to ensure uniqueness automatically. I like the former more.

**Comment by wlandau** [comment]
Fixed in 39d84ffa981e595f5fd66ed2c90198ec4e1b4cea.

**Comment by wlandau** [comment]
``` r
library(drake)
drake_plan(
  trace = TRUE,
  a = target(x, transform = map(x = c(1, 1))),
  b = target(f(a), transform = map(a))
)
#> # A tibble: 4 x 5
#>   target  command  x     a     b      
#>   <chr>   <chr>    <chr> <chr> <chr>  
#> 1 a_1     1        1     a_1   <NA>   
#> 2 a_1.1   1        1     a_1.1 <NA>   
#> 3 b_a_1   f(a_1)   1     a_1   b_a_1  
#> 4 b_a_1.1 f(a_1.1) 1     a_1.1 b_a_1.1
```

<sup>Created on 2019-01-22 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>
